pages/docs/couple-your-code/couple-your-code-defining-mesh-connectivity.md

```diff
@@ -95,3 +95,156 @@ void setMeshTriangles(int meshID, int size, int* vertices);
 void setMeshQuads(int meshID, int size, int* vertices);
 void setMeshTetrahedra(int meshID, int size, int* vertices);
 ```
+
+## Putting it all together
+
+Solvers may give you a range of information regarding vertices and faces.
+Some may give you unique ids for vertices, some only coordinates.
+In this section, we handle some common cases and how to implement them.
+
+### Solver provides IDs
+
+Your solver gives each vertex a unique identifier.
+These identifiers are also available when iterating over faces.
+
+In this case you can save a mapping from Solver ID to preCICE Vertex ID, after defining the vertices.
+When iterating over the faces, get the vertex identifiers of defining points.
+For triangular faces, these would be the 3 corner points.
+Then map these Solver IDs to preCICE IDs, and use those to define your connectivity.
+
+```cpp
+SolverInterface participant(...);
+auto meshID = participant.getMesh(...);
+
+// Define the map from the solver to the preCICE vertex ID
+std::map<Solver::VertexID, precice::VertexID> idMap;
+for (auto& vertex: solver.vertices) {
+  auto vertexID = participant.setMeshVertex(meshID, vertex.coords);
+  // set the vertexID as label
+  idMap.emplace(vertex.id, vertexID);
+}
+
+for (auto& tri: solver.triangularFaces) {
+  // Lookup the preCICE vertex ID using the solver vertex ID
+  auto a = idMap.at(tri.a.id);
+  auto b = idMap.at(tri.b.id);
+  auto c = idMap.at(tri.c.id);
+  // Then define the connectivity
+  participant.setMeshTriangle(meshID, a, b, c);
+}
+```
+
+### Solver supports custom attributes
+
+You solver doesn't provide a unique identifier for each vertex.
+It does provide a functionality to attach some kind of customised attribute to a vertex.
+This attribute is also available when iterating over faces.
+
+Examples of such attributes:
+
+* custom tags or labels: `vertex.label = myinfo` and `myinfo = vertex.label`
+* custom key-value dictionaries: `vertex.attributes[mykey] = myvalue` and `myvalue = vertex.attributes[mykey]`
+
+In this case you can save preCICE Vertex IDs as labels directly in the solver vertices.
+Define the vertices using the preCICE API, then iterate over them and apply the preCICE vertex IDs as labels.
+When iterating over faces, get the preCICE vertex IDs from the point labels, and use those to define your connectivity.
+
+```cpp
+SolverInterface participant(...);
+auto meshID = participant.getMesh(...);
+
+for (auto& vertex: solver.vertices) {
+  auto vertexID = participant.setMeshVertex(meshID, vertex.coords);
+  vertex.label = vertexID; // set the vertexID as label
```

Comment:
We probably want to define separate labels, or extend the current ones, right? I assume that the solver would still need those labels (e.g., on which vertices lie on a coupled boundary). I am not sure I fully understand this, but it also looks fine as-is.

Reply:
I suggest leaving it as is until we have a concrete example of such a case.